libs/cardano-ledger-api/test/Test/Cardano/Ledger/Api/State/Imp/QuerySpec.hs

```diff
@@ -0,0 +1,314 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Test.Cardano.Ledger.Api.State.Imp.QuerySpec (spec) where
+
+import Cardano.Ledger.Api.State.Query (
+  CommitteeMemberState (..),
+  CommitteeMembersState (..),
+  HotCredAuthStatus (..),
+  MemberStatus (..),
+  NextEpochChange (..),
+  queryCommitteeMembersState,
+ )
+import Cardano.Ledger.BaseTypes
+import Cardano.Ledger.Coin (Coin (Coin))
+import Cardano.Ledger.Conway.Governance (
+  Committee (..),
+  ConwayGovState,
+  EraGov (..),
+  GovAction (..),
+  GovActionPurpose (..),
+  PrevGovActionId (..),
+  Voter (DRepVoter),
+  cgEnactStateL,
+  ensCommitteeL,
+ )
+import Cardano.Ledger.Conway.PParams (
+  dvtCommitteeNoConfidence,
+  dvtCommitteeNormal,
+  dvtUpdateToConstitution,
+  ppCommitteeMaxTermLengthL,
+  ppDRepVotingThresholdsL,
+  ppGovActionDepositL,
+  ppGovActionLifetimeL,
+ )
+import Cardano.Ledger.Core
+import Cardano.Ledger.Credential (Credential (KeyHashObj))
+import Cardano.Ledger.Keys (
+  KeyHash,
+  KeyRole (..),
+ )
+import Cardano.Ledger.Shelley.LedgerState
+import Data.Default.Class (Default (..))
+import Data.Foldable (Foldable (..))
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+import Lens.Micro ((&), (.~))
+import Lens.Micro.Mtl
+import Test.Cardano.Ledger.Conway.ImpTest
+import Test.Cardano.Ledger.Core.Rational (IsRatio (..))
+import Test.Cardano.Ledger.Imp.Common
+
+spec ::
+  forall era.
+  ( HasCallStack
+  , ConwayEraImp era
+  , GovState era ~ ConwayGovState era
+  ) =>
+  SpecWith (ImpTestState era)
+spec =
+  it "Committee queries" $ do
+    setPParams
+    drep <- setupSingleDRep
+    c1 <- freshKeyHash
+    c2 <- freshKeyHash
+    c3 <- freshKeyHash
+    c4 <- freshKeyHash
+    c5 <- freshKeyHash
+    c6 <- freshKeyHash
+    c7 <- freshKeyHash
+    c8 <- freshKeyHash
+    ga1 <-
+      electCommittee
+        SNothing
+        drep
+        Set.empty
+        ( Map.fromList
+            [ (c1, 12)
+            , (c2, 2)
+            , (c3, 7)
+            , (c4, 5)
+            ]
+        )
+    expectMembers Set.empty
+    expectNoFilterQueryResult Nothing
+    passEpoch >> passEpoch -- epoch 2
+    expectMembers $ Set.fromList [c1, c2, c3, c4]
```

Comment:
It is better to create common binding for expectations, since it is easier to adjust the test later if necessary. Sort of like `x` is some value and I expect the outcome to be equal to `x` after some operation, rather then `x = 1` and I expect `1` after that operation.

Naturally it doesn't always make sense, but when it does it makes things easier
```suggestion
    let newMembers = Map.fromList [(c1, 12), (c2, 2), (c3, 7), (c4, 5)]
    ga1 <- electCommittee SNothing drep Set.empty newMembers
    expectMembers Set.empty
    expectNoFilterQueryResult Nothing
    passEpoch >> passEpoch -- epoch 2
    expectMembers $ Map.keysSet newMembers
```

Reply:
Yes  i agree, it makes the intention of what is tested clearer too. Thanks